pdl-compiler/src/ast.rs

```diff
@@ -143,6 +143,10 @@ pub enum FieldDesc {
     },
     #[serde(rename = "scalar_field")]
     Scalar { id: String, width: usize },
+    /// Special case of Scalar for fields used as condition for
+    /// optional fields. The width is always 1.
```

Comment:
How come the width is 1?

Reply:
Simplifies the question of what the value should be if the field is not set.
This lets the generator hide the flags in the generator since the value can be derived entirely from the optional field.